integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateTableHelper.scala

```diff
@@ -59,8 +60,22 @@ case class PreAggregateTableHelper(
     val df = sparkSession.sql(updatedQuery)
     val fieldRelationMap = PreAggregateUtil.validateActualSelectPlanAndGetAttributes(
       df.logicalPlan, queryString)
+    val partitionInfo = parentTable.getPartitionInfo
     val fields = fieldRelationMap.keySet.toSeq
     val tableProperties = mutable.Map[String, String]()
+    val childPartitionColumns = if (partitionInfo != null &&
+                              partitionInfo.getPartitionType == PartitionType.NATIVE_HIVE) {
+      val parentPartitionColumns = partitionInfo.getColumnSchemaList.asScala.map(_.getColumnName)
+      PreAggregateUtil
+        .extractPartitionInfoForAggregateTable(df.logicalPlan, parentPartitionColumns)
+    } else {Seq()}
+    val partitionerFields = fieldRelationMap.collect {
+      case (field, dataMapField) if childPartitionColumns
+        .exists(childCol =>
+          childCol.equals(dataMapField.columnTableRelationList.get.head.parentColumnName) &&
+          dataMapField.aggregateFunction.isEmpty) =>
+      PartitionerField(field.name.get, field.dataType, field.columnComment)
+    }.toSeq
```

Comment:
The order of partition columns is important. Please make sure that parent partition column order and child partition column order is same.

Reply:
PLease add some tests to make sure the order of partition columns are same as parent partition columns

Reply:
done